sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -378,6 +416,67 @@ object DateTimeUtils {
       throw QueryExecutionErrors.cannotCastUTF8StringToDataTypeError(s, TimestampType)
     }
   }
+
+  /**
+   * Trims and parses a given UTF8 string to a corresponding [[Long]] value which representing the
+   * number of microseconds since the epoch. The result is independent of time zones,
+   * which means that zone ID in the input string will be ignored.
+   * The return type is [[Option]] in order to distinguish between 0L and null. The following
+   * formats are allowed:
+   *
+   * `yyyy`
+   * `yyyy-[m]m`
+   * `yyyy-[m]m-[d]d`
+   * `yyyy-[m]m-[d]d `
+   * `yyyy-[m]m-[d]d [h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   * `yyyy-[m]m-[d]dT[h]h:[m]m:[s]s.[ms][ms][ms][us][us][us][zone_id]`
+   *
+   * where `zone_id` should have one of the forms:
+   *   - Z - Zulu time zone UTC+0
+   *   - +|-[h]h:[m]m
+   *   - A short id, see https://docs.oracle.com/javase/8/docs/api/java/time/ZoneId.html#SHORT_IDS
+   *   - An id with one of the prefixes UTC+, UTC-, GMT+, GMT-, UT+ or UT-,
+   *     and a suffix in the formats:
+   *     - +|-h[h]
+   *     - +|-hh[:]mm
+   *     - +|-hh:mm:ss
+   *     - +|-hhmmss
+   *  - Region-based zone IDs in the form `area/city`, such as `Europe/Paris`
+   *
+   * Note: The input string has to contains year/month/day fields, otherwise Spark can't determine
+   *       the value of timestamp without time zone.
+   */
+  def stringToTimestampWithoutTimeZone(s: UTF8String): Option[Long] = {
+    val (segments, tz, justTime) = parseTimestampString(s)
+    // If the input string can't be parsed as a timestamp, or it contains only the time part of a
+    // timestamp and we can't determine its date, return None.
+    if (segments.isEmpty || justTime) {
+      return None
+    }
+    try {
+      // We need to validate the time zone if exists
```

Comment:
to avoid duplicated code, shall we just ask `parseTimestampString` to return ZoneId instead of string?

Reply:
Then in `stringToTimestamp`,  we have to pass the session `timeZoneId ` to method `parseTimestampString` and assert the result `timeZoneId` is defined, which is also quite tricky.

Reply:
Why? It can return Option[ZoneId]

Reply:
Updated. I just felt that the input/output of the method would become more complicated. But I am fine with either way.